.github/dependabot.template.yml

```diff
@@ -0,0 +1,36 @@
+#! Use https://carvel.dev/ytt/ to generate this template.
+#! Either use the "Regenerate dependabot.yml" task in Codespaces, or run the following command:
+#! > ytt -f .github/dependabot.template.yml > .github/dependabot.yml
+
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    target-branch: "main"
+#@ for branch in ["main", "release/7.x", "release/7.0", "release/6.x"]:
+  - package-ecosystem: "nuget"
+    directory: "/eng/dependabot"
+    schedule:
+      interval: "daily"
+    target-branch: #@ branch
+    ignore:
+      - dependency-name: "Microsoft.Extensions.*"
+        update-types: [ "version-update:semver-major" ]
+  - package-ecosystem: "nuget"
+    directory: "/eng/dependabot/public-feed"
```

Comment:
Does "public feed" mean nuget.org? If so, maybe rename it to "nuget.org" so it's more clear as to where its getting package information. When I read/hear "public-feed" in terms of the dotnet repos, I think of https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-public